rocky/rocky/views/mixins.py

```diff
@@ -224,8 +243,22 @@ def __getitem__(self, key: Union[int, slice]) -> List[HydratedFinding]:
                 offset=offset,
                 limit=limit,
             ).items
-            ooi_references = {finding.ooi for finding in findings}
-            finding_type_references = {finding.finding_type for finding in findings}
+            if self.show_muted:
+                findings = self.get_muted_findings(findings, offset, limit)
+
+            ooi_references = set()
+            finding_type_references = set()
+            filtered_findings = []
+            for finding in findings:
+                ooi_references.add(finding.ooi)
+                finding_type_references.add(finding.finding_type)
+                for finding_type in self.finding_types:
+                    if finding_type in str(finding.finding_type):
+                        filtered_findings.append(finding)
```

Comment:
This can be a single list comprehension

Reply:
This would also need some way of keeping track of the limit/offset numbers, as those pagination calls wont match the filtered result set anymore.

Reply:
Yeah I think I need to pour it again in a Paginated list

Reply:
I have tested it and the counting and limiting happens before already. I keep re-using and mutating what octopoes gave back as `listed_findings()`